### ISSUE TITLE
Save tracked object once

### ISSUE THREAD

**Issue by spinxi** [ultralytics/ultralytics#5903]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

So, I need something like this, when yolo detects object with id 1 (not a class id but an object id itself) I need to save that frame, but only once.

So I think that this would be good idea to save object's id in a list when it's detected, image will be saved, and after that I will check if the same id is in the list then continue, otherwise, save images again if there is not that kinda id.

I need help if I'm on the right way, and how can I get that current object's id from kinda this results

`results = model.track(frame, persist=True)`

I'm trying to get that with this

```
class_ids = results[0].boxes.id.numpy()
print(class_ids)
```
but that returns list of id's and I need current id of an object, I guess.

**Comment by glenn-jocher** [comment]
@spinxi thanks for sharing your question.

The `results` object returned after using the `track` function on a model is an instance of the `Detections` class, and it indeed contains the object IDs you're interested in. The `Detections` class has an attribute `boxes`, which is a tensor containing the attributes of each detected object, with each row representing one object.

The `id` attribute of a box is the unique identifier for the detected object across different frames, which should correspond to what you're referring to as "object id".

When you execute `class_ids = results[0].boxes.id.numpy()`, the return value is actually a list of object IDs for all detected objects in the current frame. You're already on the right track!

If you want to save an image with only one specific object of interest, you can use an `if` statement to select the particular object ID. After saving it for the first time, you can then track it in a list or other data structure to make sure you indeed only save it once. 

Remember, the `results` also includes predicted bounding boxes, classes, and scores for all detected objects which you can manipulate and use for your specific needs. 

So overall, you're heading in the right direction with your current approach. If you have any additional questions or run into any issues, feel free to ask.

**Comment by spinxi** [comment]
I need something like this, (we are only interested in person class , that's not a problem at all, we have trained data with only one class)
on Live stream, appears person 1, we saved only 1 image, this person is still in the Region of interest, but we use tracking, so its id is for example 1, but we already took a pic of that person with this ID so we dont need second image, this person dissapeared, comes second person with id = 2, now we want single image for that new person with new id,
I think I describe it in details what exactly I need.

**Comment by glenn-jocher** [comment]
@spinxi from your description, it sounds like your intention is to capture and save a single image of each unique person detected in a live stream. 

In essence, you want to leverage object tracking along with object detection, such that each distinct person is assigned a unique ID, and an image of that person is saved only the first time they are detected in the frames. 

One idea is to maintain a record of the unique IDs you've encountered. When a new set of detection results is returned, you can iterate over these results, extract the object IDs, and compare them with the stored records. If a match is not found, that indicates it's a new person (because the current ID is not in the record), so you can then save that frame as an image. Then, add this new ID to your record to prevent saving frames of this person in the future.

In summary, using object detection in tandem with object tracking should allow you to achieve your goal of saving only one image per unique person detected on a live stream.
